src/rpc/net.cpp

```diff
@@ -300,6 +301,9 @@ static RPCHelpMan addnode()
     if (strCommand == "onetry")
     {
         CAddress addr;
+        if (!request.params[2].isNull()) {
+            addr.nServices = ServiceFlags{static_cast<uint64_t>(request.params[2].get_int64())};
```

Comment:
Why only for onetry connections?

Reply:
I just happened to be using it like that, but, you're right - it should work for the "add" sub-command as well. Fixed.